app/src/pages/playground/playgroundUtils.ts

```diff
@@ -1052,15 +1053,37 @@ export const getChatCompletionOverDatasetInput = ({
  * @param content - the content to normalize
  * @returns a normalized json string
  */
-export function normalizeMessageAttributeValue(
-  content?: string | null | Record<string, unknown>
-): string {
+export function normalizeMessageContent(content?: unknown): string {
   if (content == null || content === "") {
     return "{}";
   }
-  return typeof content === "string"
-    ? content
-    : JSON.stringify(content, null, 2);
+
+  if (typeof content === "string") {
+    try {
+      // If it's a double-stringified value, parse it twice
+      if (
+        content.startsWith('"{') ||
+        content.startsWith('"[') ||
+        content.startsWith('"\\"')
+      ) {
```

Comment:
Rather than putting this inline you can make it more human readable
```suggestion
      // If it's a double-stringified value, parse it twice
      if (isDoubleStringified(content)) {
```

Reply:
Some tests could be good too - non blocking

Reply:
I added tests in the diff `playgroundUtils.test.ts`